programs/bpf_loader/src/syscalls/cpi.rs

```diff
@@ -234,7 +234,9 @@ impl<'a, 'b> CallerAccount<'a, 'b> {
                 // _yet_, but we will be able to once the caller returns.
                 &mut []
             } else {
-                data_slice
+                ptr_box
```

Comment:
Makes sense, we only translate to a mutable version when we know we need it later. In direct mapping mode, we skip this.

Reply:
Yeah, it's doing an extra translation now, but that's not exactly expensive, and it's only in the mode that's essentially unused anyway...